src/flake8_bugbear/plugins/redundant_tuple_in_exception_handler.rs

```diff
@@ -1,20 +1,85 @@
-use rustpython_ast::{Excepthandler, ExcepthandlerKind, ExprKind};
+use anyhow::Result;
+use log::error;
+use rustpython_ast::{Excepthandler, ExcepthandlerKind, ExprKind, Located};
+use rustpython_parser::lexer;
+use rustpython_parser::lexer::Tok;
 
+use crate::ast::helpers;
 use crate::ast::types::Range;
+use crate::autofix::Fix;
 use crate::check_ast::Checker;
 use crate::checks::{Check, CheckKind};
+use crate::code_gen::SourceGenerator;
+use crate::SourceCodeLocator;
+
+/// Given a statement like `except (ValueError,)`, find the range of the
+/// parenthesized expression.
+fn match_tuple_range<T>(located: &Located<T>, locator: &SourceCodeLocator) -> Result<Range> {
+    // Extract contents from the source code.
+    let range = Range::from_located(located);
+    let contents = locator.slice_source_code_range(&range);
+
+    // Find the left (opening) and right (closing) parentheses.
+    let mut location = None;
+    let mut end_location = None;
+    let mut count: usize = 0;
+    for (start, tok, end) in lexer::make_tokenizer(&contents).flatten() {
+        if matches!(tok, Tok::Lpar) {
+            if count == 0 {
+                location = Some(helpers::to_absolute(&start, &range.location));
+            }
+            count += 1;
+        }
+
+        if matches!(tok, Tok::Rpar) {
+            count -= 1;
+            if count == 0 {
+                end_location = Some(helpers::to_absolute(&end, &range.location));
+                break;
+            }
+        }
+    }
+    if let (Some(location), Some(end_location)) = (location, end_location) {
+        Ok(Range {
+            location,
+            end_location,
+        })
+    } else {
+        Err(anyhow::anyhow!(
+            "Unable to find left and right parentheses."
+        ))
+    }
+}
```

Comment:
@charliermarsh Nice! Can we use this approach to find the range of a class name?

```python
class MyClass(...):
      ^^^^^^^
      This
```

Reply:
Yeah probably!

I'm also wondering if `Ident` fields like the `ClassDef` name should have locations added to them in the parser...